### PR TITLE
Add extra warning info for lr

### LINKED ISSUE #44293: tf.keras.experimental.CosineDecay error
- TensorFlow version: 2.3.0
- Python version: 3.7.6

According to https://www.tensorflow.org/api_docs/python/tf/keras/experimental/CosineDecay :
> You can pass this schedule directly into a tf.keras.optimizers.Optimizer as the learning rate. 

However when I try to use the suggested code, I get the following error:

`TypeError: '<' not supported between instances of 'CosineDecay' and 'int'`

**Standalone code to reproduce the issue**

Here is a sample code:

```
import numpy as np
from tensorflow import keras
from tensorflow.keras import layers
import tensorflow as tf

# Model / data parameters
num_classes = 10
input_shape = (28, 28, 1)

# the data, split between train and test sets
(x_train, y_train), (x_test, y_test) = keras.datasets.mnist.load_data()

# Scale images to the [0, 1] range
x_train = x_train.astype("float32") / 255
x_test = x_test.astype("float32") / 255
# Make sure images have shape (28, 28, 1)
x_train = np.expand_dims(x_train, -1)
x_test = np.expand_dims(x_test, -1)
print("x_train shape:", x_train.shape)
print(x_train.shape[0], "train samples")
print(x_test.shape[0], "test samples")


# convert class vectors to binary class matrices
y_train = keras.utils.to_categorical(y_train, num_classes)
y_test = keras.utils.to_categorical(y_test, num_classes)

model = keras.Sequential(
    [
        keras.Input(shape=input_shape),
        layers.Conv2D(64, kernel_size=(3, 3), activation="relu"),
        layers.MaxPooling2D(pool_size=(2, 2)),
        layers.Flatten(),
        layers.Dropout(0.5),
        layers.Dense(num_classes, activation="softmax"),
    ]
)

model.summary()

decay_steps = 10
lr_decayed_fn = tf.keras.experimental.CosineDecay(0.0001, decay_steps)

model.compile(
    loss='categorical_crossentropy',
    optimizer=keras.optimizers.Adam(lr=lr_decayed_fn),
    metrics=['categorical_accuracy'],
)

batch_size = 128
epochs = 5

model.compile(loss="categorical_crossentropy", optimizer=tf.keras.optimizers.Adam(lr=lr_decayed_fn), metrics=["categorical_accuracy"])

model.fit(x_train, y_train, batch_size=batch_size, epochs=epochs, validation_split=0.1)
```

### PR DESCRIPTION
**Pull request by bhack**
Fix  https://github.com/tensorflow/tensorflow/issues/44293 https://github.com/tensorflow/tensorflow/issues/44172

### CODE REVIEW

File: tensorflow/python/keras/optimizer_v2/optimizer_v2.py [hunk 1]
```diff
@@ -338,6 +338,12 @@ def my_gradient_transformer(grads_and_vars):
       if k not in allowed_kwargs:
         raise TypeError("Unexpected keyword argument "
                         "passed to optimizer: " + str(k))
+      if k == "lr" and (isinstance(kwargs[k],
+                        (ops.Tensor,
+                        learning_rate_schedule.LearningRateSchedule))
+                        or callable(kwargs[k])):
```
Comment (fchollet) [inline tensorflow/python/keras/optimizer_v2/optimizer_v2.py:344]: If we intend to issue a deprecation warning for `lr` (which is fine), we can issue it unconditionally, without additional checks on the type.

Comment (bhack) [reply tensorflow/python/keras/optimizer_v2/optimizer_v2.py:344]: I don't know. Let me know if it could be a deprecation warning or not.

**Comment by fchollet** [comment]
We're adding an unconditional deprecation warning. Closing this PR since it is no longer applicable. Thanks!

**Comment by bhack** [comment]
Yes 5 months passed.

**Pull request closed without merging**